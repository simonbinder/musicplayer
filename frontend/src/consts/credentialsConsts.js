export const SAVE_SPOTIFY_ACCESS_TOKEN = "credentials/save-spotify-access-token";
export const SAVE_USER = "credentials/save-user";
export const RESET_CREDENTIALS = "credentials/reset-credentials";
