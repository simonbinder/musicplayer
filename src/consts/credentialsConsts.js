export const SAVE_SPOTIFY_ACCESS_TOKEN = "credentials/save-spotify-access-token";
