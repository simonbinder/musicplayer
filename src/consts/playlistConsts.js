export const SAVE_PLAYLISTS_INITIAL = "playlist/save-playlists-initial";
export const SAVE_PLAYLIST = "playlist/save-new-playlist";
export const REMOVE_PLAYLIST = "playlist/remove-playlist";
export const UPDATE_PLAYLIST = "playlist/update-playlist";
export const PLAY_TRACK = "playlist/play-track";
export const SHUFFLE_STATE_CHANGED = "playlist/shuffle-state-changed";
