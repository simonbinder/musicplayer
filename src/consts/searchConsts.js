export const SEARCH_VALUE_CHANGED = "search/search-value-changed";
