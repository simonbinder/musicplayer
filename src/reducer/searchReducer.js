import {
  SEARCH_VALUE_CHANGED,
  SEARCH_TRACKS_SUCCESS,
  SEARCH_TRACKS_ERROR
} from '../consts/searchConsts';

const initialState = {
  searchValue: '',
  tracks: [],
  error: '',
};

export default function searchReducer(state = initialState, action) {
  switch(action.type) {
    case SEARCH_VALUE_CHANGED:
      state.searchValue = action.payload;
      return state;
    case SEARCH_TRACKS_SUCCESS:
      state.tracks = action.payload;
      return state;
    case SEARCH_TRACKS_ERROR:
      state.error = action.payload;
      return state;
    default:
      return state;
  }
};
