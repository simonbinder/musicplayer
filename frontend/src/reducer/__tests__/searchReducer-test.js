import searchReducer from '../searchReducer';
import {SEARCH_VALUE_CHANGED, SEARCH_TRACKS_SUCCESS, SEARCH_TRACKS_ERROR} from '../../consts/searchConsts';

describe('search reducer', () => {
  it('should return the initial state', () => {
    expect(searchReducer(undefined, {})).toMatchSnapshot();
  })

  it('search entered', () => {
    expect(searchReducer({}, {
      type: SEARCH_VALUE_CHANGED,
      payload: "Test"
    })).toMatchSnapshot();
  })

  it('search successful', () => {
    expect(searchReducer({}, {
      type: SEARCH_TRACKS_SUCCESS,
      payload: "Test"
    })).toMatchSnapshot();
  })

  it('search error', () => {
    expect(searchReducer({}, {
      type: SEARCH_TRACKS_ERROR,
      payload: "Test"
    })).toMatchSnapshot();
  })
})
