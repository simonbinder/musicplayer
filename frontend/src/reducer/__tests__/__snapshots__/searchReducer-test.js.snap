// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search reducer search entered 1`] = `
Object {
  "searchValue": "Test",
}
`;

exports[`search reducer search error 1`] = `
Object {
  "error": "Test",
  "tracks": Array [],
}
`;

exports[`search reducer search successful 1`] = `
Object {
  "error": null,
  "tracks": "Test",
}
`;

exports[`search reducer should return the initial state 1`] = `
Object {
  "error": null,
  "searchValue": "",
  "tracks": Array [],
}
`;
